packages/SwingSet/test/test-state.js

```diff
@@ -127,11 +79,58 @@ test('crankBuffer fulfills storage API', t => {
   testStorage(t, crankBuffer, () => getAllState(store).kvStuff, commitCrank);
 });
 
+test('crankBuffer handles key iteration properly even with intra-crank data changes', t => {
+  const store = initSwingStore(null);
+  const { crankBuffer: s, commitCrank } = buildCrankBuffer(
+    store.kvStore,
+    createSHA256,
+  );
+
+  s.set('k8', '8');
+  s.set('k7', '7');
+  s.set('k6', '6');
+  s.set('k5', '5');
+  s.set('k4', '4');
+  s.set('k3', '3');
+  s.set('k2', '2');
+  s.set('k1', '1');
+  commitCrank();
+
+  let keys = Array.from(s.getKeys('k', 'k~'));
```

Comment:
All of these tests use `Array.from` and drain the iterator immediately, could we have some that tests the iterator-invalidation pathways too? They should include:
* doing a `set` of a new key (whether in the middle of the range or not) invalidates all existing iterators
* doing a `set` of an old key does not
* doing a `delete` of an existing key either invalidates the iterator, or does not get included in the iterator's output after the `delete`
* deleting an existing key, then re-adding it, does something sensible

Reply:
Per our discussion above, it makes sense to add tests of correct behavior when various kinds of storage mutation operations are performed while in the midst of iteration.  However, nothing in `test-state.js` pertains to the getAfter API.  The appropriates test file to look at is `test-vatstore.js`, to which I will add the appropriate new tests suggested here.